packages/block-editor/src/components/link-control/link-preview.js

```diff
@@ -132,25 +140,24 @@ export default function LinkPreview( {
 					label={ __( 'Edit link' ) }
 					onClick={ onEditClick }
 					size="compact"
+					showTooltip={ ! showIconLabels }
 				/>
 				{ hasUnlinkControl && (
 					<Button
 						icon={ linkOff }
 						label={ __( 'Remove link' ) }
 						onClick={ onRemove }
 						size="compact"
+						showTooltip={ ! showIconLabels }
 					/>
 				) }
 				<Button
 					icon={ copySmall }
-					label={ sprintf(
-						// Translators: %s is a placeholder for the link URL and an optional colon, (if a Link URL is present).
-						__( 'Copy link%s' ), // Ends up looking like "Copy link: https://example.com".
-						isEmptyURL || showIconLabels ? '' : ': ' + value.url
```

Comment:
This was added because users specifically wanted a way to view the full URL without having to edit it. 

For visually sighted users the URL is clipped when it overflows and thus users requested a way to see the full URL, even if that requires them hovering over a button.

What's the rationale for removing this? Perhaps it's this:

> Tooltips are pointless when their text is already visible within the control.

Which seems true until you realise that the full URL is visually clipped. We don't want to undo that visual clipping (it's been iterated on _several_ times) and so this was a good way to retain UX for sighted users.

Reply:
It's not only a visual thing. You are missing the point that before being the tooltip, this is the aria-label of the button.
The aria-label provides the accessible name of a control. A name control is not the place for values or states. It just has to give the control a name that tells users what the control does.

This is basically the equivalent of the button text, would you ever provide users with a HTML button like the following one? 

![Screenshot 2024-04-23 at 09 26 51](https://github.com/WordPress/gutenberg/assets/1682452/812a044a-877b-4953-99ed-685ccc728d48)

If we want to provide users the ability to see the full URL, then the tooltip is not the right place for that. I also want to mention that these are simple, basic, accessibility principles we've been repeating sinc eyears in this project but it seems designers still don't get the point.
We can't use tooltips this way. Tooltips were introduced in Gutenberg with the only specific purpose to visually expose an icon button name. They must not be used for values, states, or descriptions.

Reply:
What I've tried to do is provide you with context you are missing regarding a UX requirement of being able to perceive the entire link. If that context doesn't currently marry with best practices then it would be nice if we could work together collaboratively and constructively to devise a more suitable solution for the benefit of users.

Do you have any suggestions for how we could achieve this?

> I also want to mention that these are simple, basic, accessibility principles we've been repeating sinc eyears in this project 

I appreciate your frustration regarding accessibility principles. I looked at the component docs and noted the lack of documentation around this component and how it should be used:

https://wordpress.github.io/gutenberg/?path=/docs/components-tooltip--docs

Given that this something you find yourself repeating, I think we could work together to improve the documentation and add specific notes on accessibility linking out to resources where appropriate.

I'll hold my hand up and accept that I could be more informed about the specifics of a11y regarding tooltips, but I strongly believe that if we want consumers (especially 3rd party, non-Core team developers) to follow best practices we should be looking at providing _documentation_ alongside the component. 

That requires a collaborative effort and I'd be more than happy to work alongside you on that if you are open to it. If not then I will do it myself.

> ...but it seems designers still don't get the point.

I'm not a designer but I think it's disingenuous to paint designers in this light. Mistakes happen, it's the will to correct and improve both the code and one's own knowledge that is key. Providing documentation is also a great way to share the knowledge of individual contributors and make it more readily available to others.

Reply:
For avoidance of confusion here's an example of where being able to see the entire URL is needed. Is this linking to the correct PDF? How do I know? 

<img width="1042" alt="Screen Shot 2024-04-23 at 10 56 40" src="https://github.com/WordPress/gutenberg/assets/444434/b39adaad-3646-463b-b344-4916cafa9ef7">

Again, I'm not advocating we continue to use a pattern that provides a poor experience for users of assistive tech but I feel we owe it to our users to find a solution that accounts for this requirement before we remove it from the UI.

Reply:
I'd like to reiterate my concerns above regarding the tooltip providing important UX for sighted users.

I've re-read the thread and acknowledhge that `Copy link: {{FULL URL HERE}}` is not helpful as an accessible `name`.

<details>
<summary>Screenshot of the long name for context</summary>
<img width="1376" alt="Screen Shot 2024-10-24 at 12 32 43" src="https://github.com/user-attachments/assets/54562513-75c6-4d92-bcd6-87bdc5d54e4a">
</details>

That said whilst fixing this for one set of users, we should avoid introducing a regression for another set.

Therefore I wonder if it's possible to have a tooltip attached to the button which isn't used as the `name` for the button but which is perceivable by all users _if they choose_? What a11y roles or semantics could be used to achieve this?

Reply:
I've re-tested the implementation on this PR and I've changed my opinion. Having a very long URL in a tooltip doesn't offer a huge amount of value. It is also detrimental to the UX for a large number of users.

With the advancements to the Link UI in WordPress 6.5 it's now much easier to just click "Edit" and see the full URL in the form input field. Prior to those changes it was necessary to click several buttons to achieve the same thing.

So overall I'd be in favour of this change.